src/main/java/seedu/address/logic/commands/EditEntryCommand.java

```diff
@@ -0,0 +1,197 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.util.CliSyntax.PREFIX_AMOUNT;
+import static seedu.address.logic.parser.util.CliSyntax.PREFIX_CATEGORY;
+import static seedu.address.logic.parser.util.CliSyntax.PREFIX_DESCRIPTION;
+import static seedu.address.logic.parser.util.CliSyntax.PREFIX_TAG;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.category.Category;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.account.ActiveAccount;
+import seedu.address.model.account.entry.Amount;
+import seedu.address.model.account.entry.Description;
+import seedu.address.model.account.entry.Entry;
+import seedu.address.model.account.entry.Expense;
+import seedu.address.model.account.entry.Revenue;
+import seedu.address.model.tag.Tag;
+
+public class EditEntryCommand extends Command {
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the entry identified "
+            + "by the index number used in the expense/revenue list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_CATEGORY + "CATEGORY] "
+            + "[" + PREFIX_DESCRIPTION + "DESCRIPTION] "
+            + "[" + PREFIX_AMOUNT + "AMOUNT] "
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_CATEGORY + "revenue"
+            + PREFIX_DESCRIPTION + "selling clothes "
+            + PREFIX_AMOUNT + "20";
+
+    public static final String MESSAGE_SUCCESS = "Edited Entry: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    private Index index;
+    private EditEntryDescriptor editEntryDescriptor;
+
+    /**
+     * Creates an EditAccountCommand to edit the current {@code ActiveAccount}
+     */
+    public EditEntryCommand(Index index, EditEntryDescriptor editEntryDescriptor) {
+        requireNonNull(index);
+        this.index = index;
+        this.editEntryDescriptor = editEntryDescriptor;
+    }
+
+    @Override
+    public CommandResult execute(Model model, ActiveAccount activeAccount) throws CommandException {
+        requireNonNull(model);
+        List<? extends Entry> lastShownList;
+        if (editEntryDescriptor.getCategory().isExpense()) {
```

Comment:
try to avoid chaining methods like this due to Demeter's law. Can create a isEntryExpense() method in the EditEntryDescriptor so that it would look like "editEntryDescriptor.isEntryExpense()". Similarly, can do the same for revenue